### PR TITLE
Alex/decode order data

### PR DESCRIPTION
**Pull request by asoong**
- Updated structs, removing overall order data header
- Remove previously implemented `MockExchangeOrderHandler` which was added just to test the `ExchangeOrderHandler` library
- Parse header as part of fillOrder
- Add `ExchangeDispatcher` as an Core extension to manage valid exchanges that can be used to fill the issuanceOrder

### CODE REVIEW

File: contracts/external/LibBytes.sol [hunk 1]
```diff
@@ -179,7 +179,7 @@ library LibBytes {
             "FROM_LESS_THAN_TO_REQUIRED"
         );
         require(
-            to < b.length,
+            to <= b.length,
```
Comment (asoong) [inline contracts/external/LibBytes.sol]: @felix2feng did they respond to your question about this?

File: test/utils/orderWrapper.ts [hunk 2]
```diff
@@ -4,46 +4,46 @@ import * as ethUtil from "ethereumjs-util";
...
-  return ethUtil.bufferToHex(
-    ethUtil.setLengthLeft(
-      ethUtil.toBuffer(orderCount), 32)
-    );
+  const exchangeOrderDatum: Buffer[] = [];
+  _.times(orderCount, (index) => {
+    const exchange = _.sample(EXCHANGES);
+    exchangeOrderDatum.push(paddedBufferForData(exchange));
+
+    const orderLength = _.random(120, 160)
```
Comment (asoong) [inline test/utils/orderWrapper.ts]: Hoping to sub any kind of util function we write later that generates a real exchange order that mirrors what `setProtocol.js` will return

File: contracts/core/extensions/CoreExchangeDispatcher.sol [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    limitations under the License.
+*/
+
+pragma solidity 0.4.24;
+
+import { Ownable } from "zeppelin-solidity/contracts/ownership/Ownable.sol";
+import { CoreState } from "../lib/CoreState.sol";
+
+
+// /**
```
Comment (felix2feng) [inline contracts/core/extensions/CoreExchangeDispatcher.sol]: Why are there 2 types of comments here?

File: contracts/core/lib/CoreState.sol [hunk 4]
```diff
@@ -29,6 +29,9 @@ contract CoreState {
     /* ============ Structs ============ */
 
     struct State {
+        // Mapping of exchagne enumeration to address
```
Comment (felix2feng) [inline contracts/core/lib/CoreState.sol]: "exchange"

File: contracts/core/lib/ExchangeHandler.sol [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+ * @author Set Protocol
+ *
+ * This library contains functions and structs to assist with parsing exchange orders data
+ */
+library ExchangeHandler {
+
+    // ============ Structs ============
+
+    struct OrderHeader {
+        uint8 exchange;
```
Comment (felix2feng) [inline contracts/core/lib/ExchangeHandler.sol]: FYI - Currently, we don't get any storage savings from packing with just a `uint8`

**Pull request merged**